scripts/parser/backend/cmakecacheparser.py

```diff
@@ -0,0 +1,37 @@
+from os.path import dirname
+from cmakecachevariable import cmakecache_variable as cmc_var
+from cmakecachestring import cmakecache_string as cmc_str
+
+class cmake_cache_parser:
+   def parse(self, file_path: str) -> list[cmc_var]:
+      """
+      this will communicate with the gui business logic, and pass to it
+      variables ready to be displayed
+      """
+      comment: str = ""
+      self.file_lines = open(dirname(__file__) + file_path).readlines()
+
+      all_vars: list[cmc_var] = []
+      for i, line in enumerate(self.file_lines):
+         parsed_string: cmc_str = cmc_str(line)
+         if parsed_string.is_variable():
+            j = i
+            while j >=0 and cmc_str(self.file_lines[j]).is_comment():
+               comment = cmc_str(self.file_lines[j]).user_comment() +  comment
+               j -= 1
+            all_vars.append(cmc_var(line, comment, i))
+
+      return all_vars
+
+   def write(self, new_vars: list[cmc_var], file_path: str) -> None:
+      """
+      this will take the modified input from the gui and write it to file, then
+      the gui will re-read  and display the file contents in order to have an
+      updated version. Since we assume new data will come from the GUI,
+      here we assume that the file is already parsed.
+      """
+      for var in new_vars:
+         self.file_lines[var.idx] = var.serialize_line()
+
+      with open(file_path, "w+") as f:
```

Comment:
If `file_lines` becomes an array of cmakecache_row objects, you could have a method serialize() there, which will call var.serialize() in case it's a variable.